src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -1155,11 +1195,40 @@ private boolean doPrepExec(TDSWriter tdsWriter, Parameter[] params, boolean hasN
             }
         }
 
-        sendParamsByRPC(tdsWriter, params);
+        sendParamsByRPC(tdsWriter, params, bReturnValueSyntax, callRpcDirectly);
 
         return needsPrepare;
     }
 
+    /**
+     * Checks if we should call RPC directly for stored procedures
+     *
+     * @param params
+     * @return
+     * @throws SQLServerException
+     */
+    boolean callRPCDirectly(Parameter[] params) throws SQLServerException {
+        int paramCount = SQLServerConnection.countParams(userSQL);
+        return (null != procedureName && paramCount != 0 && !isTVPType(params));
+    }
+
+    /**
+     * Checks if the parameter is a TVP type.
+     *
+     * @param params
+     * @return
+     * @throws SQLServerException
+     */
+    private boolean isTVPType(Parameter[] params) throws SQLServerException {
```

Comment:
SonarQube marks line 1225 as critical, citing that a static field should not be changed in a non-static method. This method can be and should be marked static. 

Nitpick: remove exception that is never thrown.
```suggestion
    private static boolean isTVPType(Parameter[] params) {
```